src/distinguished_name.cpp

```diff
@@ -84,22 +92,52 @@ DistinguishedName DistinguishedName::fromX509Name(X509_NAME *x509)
     return builder.build();
 }
 
-auto _createTuple(const DistinguishedName &dn)
+auto _createTuple(const DistinguishedName& dn)
 {
-    return std::tie(dn.commonName(),
+    return std::make_tuple(dn.commonName(),
                     dn.countryName(),
-                    dn.localityName(),
-                    dn.stateOrProvinceName(),
-                    dn.organizationalUnitName(),
-                    dn.organizationName(),
-                    dn.pkcs9EmailAddress(),
-                    dn.serialNumber(),
-                    dn.userId());
+                     dn.localityName(),
+                     dn.stateOrProvinceName(),
+                     dn.organizationalUnitName(),
+                     dn.organizationName(),
+                     dn.pkcs9EmailAddress(),
+                     dn.serialNumber(),
+                     dn.userId(),
+                     dn.givenName());
 }
 
 bool DistinguishedName::operator==(const DistinguishedName &other) const
 {
-    return _createTuple(*this) == _createTuple(other);
+    if (_customAttributeOrderFlag) {
```

Comment:
This breaks commutativity of the `==` operator when comparing an ordered and an unordered DN.

Either you also take `other._customAttributeOrderFlag` into account or we only do the unordered comparison as suggested by you.